sympy/controlsys/lti.py

```diff
@@ -0,0 +1,1363 @@
+            self.represent[3] = zeros(1)
+
+        def two():
+            if not self.represent[0].shape[0] == self.represent[1].shape[0]:
+                raise ShapeError("Shapes of A,B,C,D must fit")
+            self.represent[2] = zeros(1, self.represent[0].shape[1])
+            self.represent[3] = zeros(1, self.represent[1].shape[1])
+
+        def three():
+            if not ((self.represent[0].shape[0] == self.represent[1].shape[0]) and
+                    (self.represent[0].shape[1] == self.represent[2].shape[1])):
+                raise ShapeError("Shapes of A,B,C,D must fit")
+            self.represent[3] = zeros(self.represent[2].shape[0],
+                                      self.represent[1].shape[1])
+
+        def default():
+            # assert that A,B,C,D have matching shapes
+            if not ((self.represent[0].shape[0] == self.represent[1].shape[0]) and
+                    (self.represent[0].shape[1] == self.represent[2].shape[1]) and
+                    (self.represent[1].shape[1] == self.represent[3].shape[1]) and
+                    (self.represent[2].shape[0] == self.represent[3].shape[0])):
+                raise ShapeError("Shapes of A,B,C,D must fit")
+
+        def transferfunction():
+            # call the private method for realization finding
+            self.represent = self._find_realization(arg[0].G, arg[0].s)
+
+            # create a block matrix [[A,B], [C,D]] for visual representation
+            self._blockrepresent = BlockMatrix([[self.represent[0], self.represent[1]],
+                                               [self.represent[2], self.represent[3]]])
+
+        try:
+            if len(arg) == 0:
+                self.represent = [None] * 4
+                zero()
+            else:
+                if isinstance(arg[0], TransferFunctionModel):
+                    transferfunction()
+
+                else:
+                    # store the argument as representation of the system, fill
+                    # in noneset args with None
+                    self.represent = [None] * 4
+                    for i, a in enumerate(arg):
+                        self.represent[i] = a
+
+                    {
+                        1: one,
+                        2: two,
+                        3: three
+                    }.get(len(arg), default)()
+
+            # create a block matrix [[A,B], [C,D]] for visual representation
+            self._blockrepresent = BlockMatrix([[self.represent[0], self.represent[1]],
+                                                [self.represent[2], self.represent[3]]])
+            return None
+
+        except TypeError:
+            raise TypeError("entries of 'representation' must be matrices")
+        except AttributeError:
+            raise TypeError("entries of 'representation' must be matrices")
+        except IndexError:
+            raise TypeError("'representation' must have at least 4 matrix-valued entries")
```

Comment:
This try block is far too broad. We should never catch `AttributeError`. Maybe just remove the try/except since it isn't doing anything useful.